include/inexor/vulkan-renderer/exceptions/vk_exception.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/exceptions/exception.hpp"
+
+#include <vulkan/vulkan_core.h>
+
+namespace inexor::vulkan_renderer::exceptions {
+
+/// @brief
+class VulkanException final : public Exception {
+
+    ///
+    [[nodiscard]] std::string get_vkresult_description(const VkResult result) const;
+
+    ///
+    [[nodiscard]] std::string get_vkresult_string(const VkResult result) const;
```

Comment:
You do not use string view?

Reply:
Not possible with `std::runtime_error`.

Reply:
The constructor has nothing to do with it?

Reply:
```cpp
VulkanException::VulkanException(const VkResult result, const std::string message)
    : Exception(message + " (" + get_vkresult_string(result) + ": " + get_vkresult_description(result) + ")") {}
```

This does not work if `get_vkresult_string` or `get_vkresult_description` are `std::string_view`.

Reply:
I'm pretty sure `std::runtime_error` will accept a string view, but it's just that `Exception` is only expecting an `std::string`. You could template `Exception`'s constructor and forward it on to `runtime_error`, or just make another overload.

Reply:
You could use `append` but yeah. String would be fine for me.

Reply:
I will try the forwarding approach.

Reply:
Ok so since `std::string_view` is not accepted, I will just keep it as `std::string`.